src/bio.c

```diff
@@ -48,13 +48,20 @@
 #include "bio.h"
 #include <fcntl.h>
 
-static char* bio_worker_title[] = {
-    "bio_close_file",
-    "bio_aof",
-    "bio_lazy_free",
+typedef struct {
+    const char *const bio_worker_title;
+    pthread_t bio_thread_id;
+    pthread_mutex_t bio_mutex;
+    pthread_cond_t bio_newjob_cond;
+    list *bio_jobs;
+} bio_worker_data;
+
+static bio_worker_data bio_workers[] = {
+    {"bio_close_file"},
+    {"bio_aof"},
+    {"bio_lazy_free"}
 };
-
-#define BIO_WORKER_NUM (sizeof(bio_worker_title) / sizeof(*bio_worker_title))
+#define BIO_WORKER_NUM (sizeof bio_workers / sizeof *bio_workers)
```

Comment:
```suggestion
#define BIO_WORKER_NUM (sizeof(bio_workers) / sizeof(*bio_workers))
```

Reply:
If `j` and `BIO_WORKER_NUM` is not of the same type, the smaller one has to be converted to the larger one in the comparison between the two, so that is a difference, with no benefit as I see it.

... and why put parentheses around the `sizeof` _expression_? It makes it look like `bio_workers` is a _type_.

Anyway, I've closed this as I don't think we'll see eye to eye on this ever.